File: Backend/src/main/java/se331/project/rest/service/FolderService.java
package se331.project.rest.service;

import org.springframework.data.domain.Page;
import se331.project.rest.entity.Announcement;
import se331.project.rest.entity.Folder;

import java.util.List;

public interface FolderService {
    Integer getFolderSize();
    Folder save (Folder folder);
    Page<Folder> getFolders(Integer pageSize, Integer page);
    List<Folder> getFolders();
    Folder getFolder(Long id);
    Folder addFolder(Folder folder);
    Folder deleteFolder(Long id);
    Folder addRecipe(Long recipeId, Long id);
    Folder deleteRecipe(Long recipeId, Long id);
//    List<Announcement> saveRecipes(List<Announcement> recipes);
}

File: Frontend/src/services/FolderService.ts
import apiClient from "./AxiosClient";
import type { AxiosResponse } from "axios";
import type { FolderItem } from "@/type";

export default {
    getFolders() : Promise<AxiosResponse<FolderItem[]>> {
        return apiClient.get<FolderItem[]>("/folders")
    },
    getFolder(id: number) : Promise<AxiosResponse<FolderItem>> {
        return apiClient.get<FolderItem>("/folder/" + id)
    },
    addFolder(name: String) : Promise<AxiosResponse<FolderItem>> {
        return apiClient.post<FolderItem>("/addFolder", { name: name })
    },
    deleteFolder(id: number) : Promise<AxiosResponse<FolderItem>> {
        return apiClient.post<FolderItem>("/deleteFolder/" + id)
    },
    addRecipe(data:Object) : Promise<AxiosResponse<FolderItem>> {
        return apiClient.post<FolderItem>("/addRecipe", data)
    },
    deleteRecipe(data: Object) : Promise<AxiosResponse<FolderItem>> {
        return apiClient.post<FolderItem>("/deleteRecipe", data)
    },
}